### PR TITLE
several build updates

### PR DESCRIPTION
**Pull request by sezero**

### CODE REVIEW

**Review by evpobr — APPROVED** [review]

File: examples/audio_out.c [hunk 1]
```diff
@@ -6,21 +6,25 @@
...
 */
 
+#ifdef HAVE_CONFIG_H
+#include "config.h"
+#endif
+
 #include <stdio.h>
 #include <stdlib.h>
 #include <string.h>
+#ifdef HAVE_UNISTD_H
```
Comment (SoapGentoo) [inline examples/audio_out.c:16]: I assume this is for Windows?

File: examples/varispeed-play.c [hunk 2]
```diff
@@ -6,16 +6,18 @@
 ** file at : https://github.com/libsndfile/libsamplerate/blob/master/COPYING
 */
 
+#ifdef HAVE_CONFIG_H
+#include "config.h"
+#endif
```
Comment (SoapGentoo) [inline examples/varispeed-play.c:11]: thank you, `config.h` should always be the first include in every `.c` file

File: src/src_zoh.c [hunk 3]
```diff
@@ -6,15 +6,15 @@
 ** file at : https://github.com/libsndfile/libsamplerate/blob/master/COPYING
 */
 
+#ifdef HAVE_CONFIG_H
+#include "config.h"
+#endif
```
Comment (SoapGentoo) [inline src/src_zoh.c:11]: make it consistent: Add an empty line after this block in every file

File: tests/clone_test.c [hunk 4]
```diff
@@ -28,7 +32,7 @@ clone_test (int converter)
 	SRC_STATE* src_state_cloned ;
 	SRC_DATA src_data, src_data_cloned ;
 
-	int error, frame, ch, index ;
+	int error, frame, ch, idx ;
```
Comment (SoapGentoo) [inline tests/clone_test.c:35]: why the change?

**Review by SoapGentoo — CHANGES_REQUESTED** [review]

File: examples/audio_out.c (hunk 1 above)
Comment (sezero) [reply examples/audio_out.c:16]: Practically, yes.

File: src/src_zoh.c (hunk 3 above)
Comment (sezero) [reply src/src_zoh.c:11]: Done.

File: tests/clone_test.c (hunk 4 above)
Comment (sezero) [reply tests/clone_test.c:35]: As mentioned in the commit message, it triggered -Wshadow and failed make distcheck on my setup.

**Review by SoapGentoo — APPROVED** [review]
@sezero looking good, could you squash related commits together? 10 commits seems like a lot for the changes here.

**Comment by sezero** [comment]
> @sezero looking good,

Thanks.

> could you squash related commits together? 10 commits seems like a lot for the changes here.

Done

**Comment by SoapGentoo** [comment]
@evpobr feel free to merge whenever you want. Also, given that libsamplerate doesn't use `autogen`, I think we should consider ditching autotools and moving to CMake only.

**Comment by sezero** [comment]
>  given that libsamplerate doesn't use `autogen`,

Hm? `autogen` as in top-level `autogen.sh` in the source tree?

> I think we should consider ditching autotools and moving to CMake only.

Sad..

**Comment by SoapGentoo** [comment]
@sezero autogen as in GNU Autogen (https://www.gnu.org/software/autogen/), it's a macro language we used in libsndfile for the test harnesses and has been a major compatibility nightmare between Unix and Windows...

> Sad..

oh are you in favour of the Autotools build system? On balance, I am usually too, because I don't like the accidental complexity CMake usually entails (All these NIH `Find*.cmake` scripts that could be simplified a lot using pkg-config). Let's shelf the idea then. It doesn't happen often that people in 2020 still cherish the Autotools...

**Comment by sezero** [comment]
> oh are you in favour of the Autotools build system?

Oh yes.

> It doesn't happen often that people in 2020 still cherish the Autotools...

Well, I'm old :)

**Comment by evpobr** [comment]
> @evpobr feel free to merge whenever you want. Also, given that libsamplerate doesn't use `autogen`, I think we should consider ditching autotools and moving to CMake only.

Actually you can merge this PR, you have the same rights in this repo as me.

**Comment by SoapGentoo** [comment]
@evpobr I know, I was just putting you in charge ;)

**Comment by evpobr** [comment]
> Also, given that libsamplerate doesn't use `autogen`, I think we should consider ditching autotools and moving to CMake only.

It's actually not very convenient to support two build systems. And if you choose between Autotools and CMake, I will of course choose the second one.

In fact, I was also thinking of removing Autotools from here. But first we will discuss everything and if we agree, we will first declare it deprecated and delete it after a few releases.

**Comment by SoapGentoo** [comment]
@evpobr the main issue I have with the current CMake buildsystem is that it is too convoluted. If you're willing to switch to `pkg-config` as a first mechanism, and then fall back to guessing flags that would be fine. Upstreams may change in weird ways, and the only thing that remains steady is the output of `pkg-config --cflags <DEP>`, whereas `Find<DEP>.cmake` modules break so often in Gentoo, that they're a real pain to maintain.

**Comment by evpobr** [comment]
I agree with that. I suggest we discuss this after the next release.
